### ISSUE TITLE
Trying to get 2-way communication and card working with Amcrest AD410

### ISSUE THREAD

**Issue by Cr4z33** [dermotduffy/frigate-hass-card#1214]
I got helped by @dermotduffy at giving a read about [2-way communication](https://github.com/dermotduffy/frigate-hass-card#using-2-way-audio) and so I tried to setup the card for it, but I get this error:
```
Impossibile ottenere l'endpoint della videocamera per questo provider live (configurazione incompleta?). [Controllare la risoluzione dei problemi](https://github.com/dermotduffy/frigate-hass-card#troubleshooting).
{
  "hide": false,
  "dependencies": {
    "all_cameras": false,
    "cameras": []
  },
  "triggers": {
    "motion": false,
    "occupancy": true,
    "entities": []
  },
  "engine": "auto",
  "frigate": {
    "client_id": "frigate"
  },
  "motioneye": {
    "images": {
      "directory_pattern": "%Y-%m-%d",
      "file_pattern": "%H-%M-%S"
    },
    "movies": {
      "directory_pattern": "%Y-%m-%d",
      "file_pattern": "%H-%M-%S"
    }
  },
  "live_provider": "go2rtc",
  "image": {
    "refresh_seconds": 1
  },
  "camera_entity": "camera.citofono_sub",
  "go2rtc": {
    "modes": [
      "webrtc"
    ]
  }
}
```
The card YAML is like the following:
```
type: custom:frigate-card
cameras:
  - camera_entity: camera.citofono_sub
    live_provider: go2rtc
    go2rtc:
      modes:
        - webrtc
view:
  dark_mode: auto
menu:
  style: overlay
  position: bottom
  buttons:
    microphone:
      enabled: true
      icon: mdi:microphone

```
Frigate (latest stable in a Docker) config is:
```
mqtt:
  host: ***.***.*.**

cameras:
  back:
    ffmpeg:
      inputs:
        - path: rtsp://admin:****@***.***.*.***:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - detect
    detect:
      width: 2560
      height: 1920
```
go2rtc (v1.5 running in the Frigate docker) config is:
```
log:
  level: info # default level
  api: trace
  exec: debug
  ngrok: info
  rtsp: debug
  stream: debug
  webrtc: debug

streams:
  # you need to include the #backchannel=0 to allow the video feed with audio (no2 way audio) on multiple devices.	
  citofono_monodirezionale: rtsp://admin:****@***.***.*.***:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
  citofono_bidirezionale: rtsp://admin:****@***.***.*.***:554/cam/realmonitor?channel=1&subtype=0
```
Welp! 🏳️

**Comment by dermotduffy** [comment]
Guess: Is `camera.citofono_sub` actually a Frigate camera entity (as exposed by the Frigate integration)?

If not, you may need to specify the Frigate camera name explicitly: 

```yaml
cameras:
  - camera_entity: camera.citofono_sub
    live_provider: go2rtc
    frigate:
      camera_name: NAME_OF_FRIGATE_CAMERA_HERE
    go2rtc:
      modes:
        - webrtc
```

**Comment by Cr4z33** [comment]
@dermotduffy  thanks I made the adjustment by replacing the camera name to what is shown in the Frigate integration, but there's no 2-way communication enabled and I noticed also that the microphone button is not shown in the menu (although I have enabled it).
```
type: custom:frigate-card
cameras:
  - camera_entity: camera.back
    live_provider: go2rtc
    frigate:
      camera_name: camera.back
    go2rtc:
      modes:
        - webrtc
      stream: citofono_bidirezionale
    webrtc_card: {}
menu:
  buttons:
    microphone:
      enabled: true
```

**Comment by dermotduffy** [comment]
@Cr4z33 I think your Frigate camera name is `back` (based on your Frigate config), not `camera.back`. Try changing that. 

 - At the very least, snapshots/clips should work in the card, do they?
 - At the very least, WebRTC video should load, does it?

If the above proposed camera name tweak doesn't work,  I highly recommend trying to get your microphone to work with go2rtc directly, and only then come back to the card. If it works for go2rtc, 99% of the work is done, and I do not see your camera listed in the list of camera that support 2-way audio (https://github.com/AlexxIT/go2rtc#two-way-audio).

**Comment by Cr4z33** [comment]
OK I changed the name to `back` and the code now looks like
```
type: custom:frigate-card
cameras:
  - camera_entity: camera.back
    live_provider: go2rtc
    frigate:
      camera_name: back
    go2rtc:
      modes:
        - webrtc
      stream: citofono_bidirezionale
menu:
  buttons:
    microphone:
      enabled: true

```
snapshots/clips work and so does WebRTC.

I actually was playing with go2rtc and its WebRTC before going to try out your card, but I was annoyed by the fact that the stream always loaded muted and I had to tap and enable the doorbell audio.

2-way audio works (if you look right under [here ](https://github.com/AlexxIT/go2rtc#source-rtsp)you will see it hehe) and it's actually a video doorbell. 😉

Just to be double sure I need to refer to go2rtc embedded in the Frigate container right?

Not the one that is installed straight into HA right? 🤔

Also I had the mic button showing in the card (before starting again from scratch and trying to get 2-way comm working), but although I have enabled it there's no way to show it.

[Screenshot](https://imgur.com/a/Gg8yidz)

**Comment by dermotduffy** [comment]
> Just to be double sure I need to refer to go2rtc embedded in the Frigate container right?

Correct.

> Not the one that is installed straight into HA right? 🤔

There is no go2rtc installed with HA, unless you manually did it yourself (but yeah -- you only need the one that comes as part of Frigate).

> Also I had the mic button showing in the card (before starting again from scratch and trying to get 2-way comm working), but although I have enabled it there's no way to show it.

It only shows up if the stream actually has 2-way audio capabilities *and* it's enabled. Since it's not showing up for you, somehow the stream is not being established correctly I'd guess.

Can you disable showing the image during load and make sure you still see the WebRTC stream loading?

```
live:
  show_image_during_load: false
```

**Comment by dermotduffy** [comment]
BTW: 2-way audio only works if you are accessing HA over `https`.

**Comment by Cr4z33** [comment]
> Can you disable showing the image during load and make sure you still see the WebRTC stream loading?
> 
> ```
> live:
>   show_image_during_load: false
> ```

I added that and now there's no way to load the stream anymore (tried both Firefox and Chrome for Windows).

https://imgur.com/a/DfdZJTv

Also yes I am using HA over https


Tell me what code would you exactly enter for the card if you had a 2-way camera/doorbell/whatever and you would like to have what I want to have? 😄 

Perhaps I am missing some crucial part hehe.

**Comment by dermotduffy** [comment]
> I added that and now there's no way to load the stream anymore (tried both Firefox and Chrome for Windows).

This is your problem, the go2rtc stream is not loading (and that's why the microphone button is not appearing). When you are not using `show_image_during_load` there's a refreshing image that is making it seem like your stream is working.

You need to figure out why your go2rtc stream is not working.

> Tell me what code would you exactly enter for the card if you had a 2-way camera/doorbell/whatever and you would like to have what I want to have? smile

There is nothing wrong with your card config :-)

**Comment by dermotduffy** [comment]
> If I get an image then it's supposed to show also the mic button right?

It will show the microphone button once the stream is established (i.e. video is visible) and if the stream supports audio (i.e. meets the [requirements](https://github.com/dermotduffy/frigate-hass-card#using-2-way-audio)).

**Comment by dermotduffy** [comment]
As [above](https://github.com/dermotduffy/frigate-hass-card/issues/1214#issuecomment-1614840747) my recommendation remains to get 2-way audio working with go2rtc alone, then once it's working come back to the card. If you can't get it working with go2rtc, it will never work with the card.

**Comment by Cr4z33** [comment]
With go2rtc it works. :)

I use the generated WebRTC link (with option "video+audio+microphone" of course).

**Comment by dermotduffy** [comment]
> With go2rtc it works. :)

Great, that means it really should work just fine in the card. No idea why your go2rtc stream won't load for you. Anything interesting in the Javascript console (press F12 in Chrome)? Can you hold down the Frigate menu button for a few seconds and paste the diagnostics you get into this issue?

**Comment by Cr4z33** [comment]
You were correct I use the generated WebRTC link (with option "video+audio+microphone" of course).

With the Supervisor go2rtc integration it works perfectly while with the version in the Frigate container not!

What I don't understand is they are same version and same config so I need to investigate further and will repost once sorted.

In the meantime thank you for your support!

**Comment by dermotduffy** [comment]
> With the Supervisor go2rtc integration it works perfectly while with the version in the Frigate container not!

Note that as of right now the card will only work with the go2rtc that comes with Frigate. [This feature request](https://github.com/dermotduffy/frigate-hass-card/issues/1206) would change that. But from your earlier post looks like you _are_ using the go2rtc in the Frigate container.

**Comment by Cr4z33** [comment]
Yes correct.

**Comment by dermotduffy** [comment]
I'm not sure there is work for the card to do on this issue -- so perhaps we should close it. If you think I'm wrong, let me know!

**Comment by Cr4z33** [comment]
Yeah sorry for being late, but I found a way to use 2-way comm by merging different HA solutions.

**Comment by BlackBeltPanda** [comment]
> Yeah sorry for being late, but I found a way to use 2-way comm by merging different HA solutions.

Can you explain your solution? Also running into issues getting 2-way audio working with the AD410.

**Comment by dermotduffy** [comment]
@BlackBeltPanda If 2-way audio works fine in go2rtc, there is absolutely no reason it should not work in the card. Perhaps open a new issue and lets discuss?

**Comment by SusithAbeythunga** [comment]
> camera_name: back

Did you find a solution?
